RDMA: Support IMB-EXT & IMB-IO in Intel MPI Benchmark

juhlee-microsoft opened the pull request:
Added IMB-EXT and IMB-IO tests from Intel MPI Benchmark. 

### Test results

iamshital commented:
Execution is verified.

`[INFO ] LISAv2 exits with code: <no exit code>` changed to `[INFO ] LISAv2 exits with code: 0`

ader1990 commented on RunTests.ps1:
```diff
@@ -438,7 +438,7 @@ try {
 		Set-Location $originalWorkingDirectory
 	}
-	Get-Variable -Exclude PWD,*Preference | Remove-Variable -Force -ErrorAction SilentlyContinue
+	Get-Variable -Exclude PWD,*Preference,ExitCode | Remove-Variable -Force -ErrorAction SilentlyContinue
```
when run manually, this line leaves your environment broken in case you might have had special env variables.
I suggest you to create a inclusive list with all the lisav2 env variables and only remove those.

ader1990 replied:
If you do this, it will help "documenting" all of these globals other scripts expect to find.
All in all, using globals is a major lisav2 flaw that needs to be addressed by using function parameters and encapsulation.

iamshital replied:
The original line was - 
`Get-Variable -Scope Global | Remove-Variable -Force -ErrorAction SilentlyContinue`
This keeps the session clean for the next execution. LISAv2 doesn't break when running another iteration in same PS session (and replaces any runtime variables).
But, later, above was changed to -
`Get-Variable -Exclude PWD,*Preference | Remove-Variable -Force -ErrorAction SilentlyContinue`
The removes all the variables from PS session leaving PS session broken for next LISAv2 execution.
Reference: https://github.com/LIS/LISAv2/commit/1a5f69334356fb8a5431f286d1b7cf87d8157b17#diff-59264e449e1f333881e7c9485a2e7a0f

If we want to fix the broken PS session issue, then we can revert it to original line.
The scope of this PR is to fix LISAv2 exit code only.

I agree with documenting on all the Global Variables. But, I also have plan to encapsulate all the global variables in single PS object. We can discuss this over email.

iamshital commented:
Opening two new issues as per discussion in this thread -
https://github.com/LIS/LISAv2/issues/148
https://github.com/LIS/LISAv2/issues/149

juhlee-microsoft committed:
Intel Benchmark P2P support

juhlee-microsoft committed:
Added P2P verification string in output

juhlee-microsoft committed:
Fxied variable name for intel MPI

modified_slaves is only available in IBM Platform MPI

juhlee-microsoft committed:
Support IMB-EXT tests

juhlee-microsoft committed:
Support IMB-IO in intel MPI Benchmark

juhlee-microsoft committed:
Fixed typo in logmsg

juhlee-microsoft committed:
Merge branch 'master' into juhlee/io_support

juhlee-microsoft committed:
Remove mpi-benchmark installation in Intel MPI

Intel MPI package provides IMB-MPI1, IMB-NBC and IMB-RMA binaries only. Removed P2P and other future tests enablement in Intel MPI section.

juhlee-microsoft committed:
Merge branch 'juhlee/use_bin_intel_mpi' into juhlee/io_support

juhlee-microsoft committed:
Merge branch 'master' into juhlee/io_support
